### PR TITLE
fix: information disclosure bug in preconditions GET

### PR DESCRIPTION
**Pull request by harshavardhana**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
fix: information disclosure bug in preconditions GET

## Motivation and Context
Pre-condition check was being honored before, validating if 
anonymous access is allowed on the metadata of an object, 
leading to metadata disclosure of the following headers.

```
Last-Modified
Etag
x-amz-version-id
Expires:
Cache-Control:
```

Although the information presented is minimal and opaque, 
it still simply discloses whether an object by a specific 
The name exists, even without such permissions.

## How to test this PR?
TBD (will update soon)

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Review by shtripat — APPROVED** [review]
LGTM

File: cmd/object-handlers.go [hunk 1]
```diff
@@ -476,6 +476,15 @@ func (api objectAPIHandlers) getObjectHandler(ctx context.Context, objectAPI Obj
 			return true
 		}
 
+		if oi.UserTags != "" {
```
Comment (vadmeste) [inline cmd/object-handlers.go:479]: Shouldn't this block come after authorizeRequest(), so we do not set any header in the access denied response

Comment (harshavardhana) [reply cmd/object-handlers.go:479]: no @vadmeste w.Header() is for the response r.Header() is incoming request

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**